test/js/node/test/parallel/test-http-outgoing-proto.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+const http = require('http');
+const OutgoingMessage = http.OutgoingMessage;
+const ClientRequest = http.ClientRequest;
+const ServerResponse = http.ServerResponse;
+
+assert.strictEqual(
+  typeof ClientRequest.prototype._implicitHeader, 'function');
+assert.strictEqual(
+  typeof ServerResponse.prototype._implicitHeader, 'function');
+
+// validateHeader
+assert.throws(() => {
+  const outgoingMessage = new OutgoingMessage();
+  outgoingMessage.setHeader();
+}, {
+  code: 'ERR_INVALID_HTTP_TOKEN',
+  name: 'TypeError',
+  message: 'Header name must be a valid HTTP token ["undefined"]'
+});
+
+assert.throws(() => {
+  const outgoingMessage = new OutgoingMessage();
+  outgoingMessage.setHeader('test');
+}, {
+  code: 'ERR_HTTP_INVALID_HEADER_VALUE',
+  name: 'TypeError',
+  message: 'Invalid value "undefined" for header "test"'
+});
+
+assert.throws(() => {
+  const outgoingMessage = new OutgoingMessage();
+  outgoingMessage.setHeader(404);
+}, {
+  code: 'ERR_INVALID_HTTP_TOKEN',
+  name: 'TypeError',
+  message: 'Header name must be a valid HTTP token ["404"]'
+});
+
+assert.throws(() => {
+  const outgoingMessage = new OutgoingMessage();
+  outgoingMessage.setHeader.call({ _header: 'test' }, 'test', 'value');
+}, {
+  code: 'ERR_HTTP_HEADERS_SENT',
+  name: 'Error',
+  message: 'Cannot set headers after they are sent to the client'
+});
+
+assert.throws(() => {
+  const outgoingMessage = new OutgoingMessage();
+  outgoingMessage.setHeader('200', 'あ');
+}, {
+  code: 'ERR_INVALID_CHAR',
+  name: 'TypeError',
+  message: 'Invalid character in header content ["200"]'
+});
+
+// // write
+// {
+//   const outgoingMessage = new OutgoingMessage();
+
+//   assert.throws(
+//     () => {
+//       outgoingMessage.write('');
+//     },
+//     {
+//       code: 'ERR_METHOD_NOT_IMPLEMENTED',
+//       name: 'Error',
+//       message: 'The _implicitHeader() method is not implemented'
```

Comment:
Not sure if node intentionally does not implement `_implicitHeader` or if they just... didn't implement it

Reply:
Please avoid modifying Node.js tests unnecessarily.